### PR TITLE
remove Model.scala

### PR DESCRIPTION
**Pull request by tpolecat**
This removes `seqexec.model.Model` … all the inner classes are now top-level and the type aliases are in a package object.

### CODE REVIEW

File: modules/seqexec/model/src/main/scala/seqexec/model/package.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import cats._
+import seqexec.model.enum._
+
+package object model {
+  type ParamName = String
+  type ParamValue = String
+  type Parameters = Map[ParamName, ParamValue]
+  type StepConfig = Map[SystemName, Parameters]
+  implicit val stEq: Eq[StepConfig] = Eq.fromUniversalEquals
+  type StepId = Int
```
Comment (cquiroz) [inline modules/seqexec/model/src/main/scala/seqexec/model/package.scala]: Would you switch this for `Index`?

File: modules/seqexec/model/src/main/scala/seqexec/model/package.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  type ObservationName = String
+  type TargetName = String
```
Comment (cquiroz) [inline modules/seqexec/model/src/main/scala/seqexec/model/package.scala]: Observation and Target name probably don't have a type in gem. are they just strings?

File: modules/seqexec/model/src/main/scala/seqexec/model/package.scala (hunk 1 above)
Comment (tpolecat) [reply modules/seqexec/model/src/main/scala/seqexec/model/package.scala]: `Index` is a positive short, so yeah anything where you start numbering at 1.

File: modules/seqexec/model/src/main/scala/seqexec/model/package.scala (hunk 2 above)
Comment (tpolecat) [reply modules/seqexec/model/src/main/scala/seqexec/model/package.scala]: Yeah they're just strings.

**Pull request merged**